src/transformers/trainer.py

```diff
@@ -4606,6 +4606,11 @@ def create_accelerator_and_postprocess(self):
         # some Trainer classes need to use `gather` instead of `gather_for_metrics`, thus we store a flag
         self.gather_function = self.accelerator.gather_for_metrics
 
+        if "use_gather_object" in inspect.signature(self.gather_function).parameters.keys():
+            self.gather_function = functools.partial(
+                self.gather_function, use_gather_object=self.args.eval_use_gather_object
+            )
```

Comment:
Let's not rely on `inspect` and just use `is_accelerate_available("0.30.0")` please!

Reply:
@muellerzr There is a test that self.gather_function has no `use_gather_function` even though accelerate_version is larger than 0.30.0. Because self.gather_function can be replaced by the trainer. such as accelerator.gather() not accelerator.gather_for_metrics().

https://app.circleci.com/pipelines/github/huggingface/transformers/96173/workflows/724facf7-3a67-4f89-81bd-0e389a775717/jobs/1268543

If there is further discussion we can talke here https://github.com/huggingface/transformers/pull/31799 !